tests/integration_tests/test_llm.py

```diff
@@ -628,23 +780,35 @@ def test_load_pretrained_adapter_weights(adapter):
 
 def _compare_models(model_1: torch.nn.Module, model_2: torch.nn.Module) -> bool:
     # For a full explanation of this 8-bit workaround, see https://github.com/ludwig-ai/ludwig/pull/3606
-    def filter_for_weight_format(i):
-        """Remove bitsandbytes metadata keys added on state dict creation.
+    # def filter_for_weight_format(i):
+    #     """Remove bitsandbytes metadata keys added on state dict creation.
+    #
+    #     8-bit quantized models that have been put on gpu will have a set of `weight_format` keys in their state dict.
+    #     These contain strings that are used to reshape quantized tensors, however these have no impact until the state
+    #     dict is loaded into a model. These keys were causing `torch.equal` to raise an exception, so we skip them in
+    #     the evaluation.
+    #     """
+    #     return "weight_format" not in i[0]
+
+    # model_1_filtered_state_dict = filter(filter_for_weight_format, model_1.state_dict().items())
+    # model_2_filtered_state_dict = filter(filter_for_weight_format, model_2.state_dict().items())
 
-        8-bit quantized models that have been put on gpu will have a set of `weight_format` keys in their state dict.
-        These contain strings that are used to reshape quantized tensors, however these have no impact until the state
-        dict is loaded into a model. These keys were causing `torch.equal` to raise an exception, so we skip them in the
-        evaluation.
-        """
-        return "weight_format" not in i[0]
+    # Source: https://discuss.pytorch.org/t/check-if-models-have-same-weights/4351/6
 
-    model_1_filtered_state_dict = filter(filter_for_weight_format, model_1.state_dict().items())
-    model_2_filtered_state_dict = filter(filter_for_weight_format, model_2.state_dict().items())
+    if model_1.__class__.__name__ != model_2.__class__.__name__:
+        return False
 
```

Comment:
I'm okay with commenting out some of these lines of code for now since we still don't have GPU tests setup for Ludwig (currently a work in progress), but this check for filtering the state dict for weight formats with 8 bit quantization is necessary to make sure the tests for comparing models works correctly when we test fine-tuning with 8-bit quantization!

Reply:
I'd maybe create a comment with a TODO here to re-enable those lines of code when GPU tests are enabled? That should be fine for now

Reply:
@arnavgarg1 The issue is that the `filter_for_weight_format()` method (which focuses on 8-bit quantization) is not currently used, and it was causing linter errors.  Could you please suggest what we should do?  Should we keep it commented out for now (I added a TODO in the updated PR), or try to re-enable it and use it?  Thank you.

Reply:
I think it should be okay to keep it commented out for now since we have the TODO - maybe I can take a look at that in a follow-up PR, but for now I don't want to block merging this awesome change into Ludwig because of it since it's causing no harm!

Reply:
cc: @jeffkinnison just a quick FYI so you're not surprised by this change in the test. I will work on adding it back when GPU tests get enabled